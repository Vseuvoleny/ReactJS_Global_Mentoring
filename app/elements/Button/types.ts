type ButtonType =
    'button' | 'reset' | 'submit'


export interface IButton {
    clickHandler?: () => void,
    type: ButtonType;
    text: string
}