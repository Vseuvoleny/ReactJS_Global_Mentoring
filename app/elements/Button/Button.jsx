import React from "react";
import cn from "classnames";
import classes from "./Buttons.module.scss";

const Button = ({ clickHandler, type, className, text }) => {
  const cls = cn({
    button: type === "button",
    reset: type === "reset",
    submit: type === "submit",
  });

  return (
    <div>
      <button
        className={`${classes.button} ${cls}`}
        type={type}
        onClick={clickHandler}
      >
        {text}
      </button>
    </div>
  );
};

export default Button;
