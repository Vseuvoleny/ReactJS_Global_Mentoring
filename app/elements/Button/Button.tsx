import React, { FC } from "react";
import cn from "classnames";
import classes from "./Buttons.module.scss";
import { IButton } from "./types";

const Button: FC<IButton> = ({ clickHandler, type, text }) => {
  const cls = cn(classes.button, {
    button: type === "button",
    reset: type === "reset",
    submit: type === "submit",
  });

  return (
    <div>
      <button className={cls} type={type} onClick={clickHandler}>
        {text}
      </button>
    </div>
  );
};

export default Button;
