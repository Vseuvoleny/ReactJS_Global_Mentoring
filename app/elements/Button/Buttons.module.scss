.button {
  outline: none;
  border: 0;
  cursor: pointer;
  text-transform: uppercase;
  border-radius: 4px;
}
