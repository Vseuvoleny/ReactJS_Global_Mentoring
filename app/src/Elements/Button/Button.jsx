import React from "react";
import "./Buttons.scss";

const Button = ({ clickHandler, type, className, text }) => {
  return (
    <div className={className}>
      <button
        className={`${className}__button`}
        type={type}
        onClick={clickHandler}
      >
        {text}
      </button>
    </div>
  );
};

export default Button;
