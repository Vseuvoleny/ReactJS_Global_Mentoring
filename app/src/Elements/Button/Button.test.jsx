import React from "react";
import { render, screen } from "@testing-library/react";
import Button from "./Button";
import userEvent from "@testing-library/user-event";

describe("Button test", () => {
  test("Button should render", () => {
    const { container } = render(<Button text="button_text" />);
    expect(screen.getByText("button_text")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test("should invoke clickHandler Fn by click", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button clickHandler={handleClick} text={"button_text"} />
    );
    const button = getByText("button_text");
    userEvent.click(button);
    expect(handleClick).toHaveBeenCalled();
  });
  test("should contain specific classes", () => {
    const { getByText } = render(
      <Button className={"classes"} type={"submit"} text={"button_text"} />
    );
    const button = getByText("button_text");
    expect(button.className).toEqual("classes__button filled");
  });
});
