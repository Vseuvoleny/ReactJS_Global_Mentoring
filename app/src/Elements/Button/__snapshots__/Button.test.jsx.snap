// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button should render 1`] = `
<div>
  <div>
    <button
      class="undefined__button filled"
    >
      button_text
    </button>
  </div>
</div>
`;
